crates/geoip/src/block.rs

```diff
@@ -49,19 +49,48 @@ impl CountryFilter {
     where
         R: Resolver,
     {
-        let country = resolver
+        let geo_data = resolver
             .lookup_geo_data_raw(addr)
-            .map_err(|_| Error::UnableToExtractGeoData)?
+            .map_err(|_| Error::UnableToExtractGeoData)?;
+
+        let country = geo_data
             .country
             .and_then(|country| country.iso_code)
             .ok_or(Error::CountryNotFound)?;
 
-        let blocked = self
-            .blocked_countries
-            .iter()
-            .any(|blocked_country| blocked_country == country);
+        let subdivisions: Vec<&str> = geo_data
```

Comment:
Let's try not to cause unnecessary allocations, since this is in a hot path.
- Subdivisions here don't need to be collected into `Vec`, returning `impl Iterator` is fine;
- `self.blocked_zones` should be pre-parsed into structs e.g. `struct BlockedZone { country: String, regions: Vec<u32> }`. Doing this on each lookup doesn't really make sense. Note: `u32` there assumes ISO code.